Invoke-SqlCmd2/Public/Invoke-SqlCmd2.ps1

```diff
@@ -441,6 +445,11 @@ function Invoke-Sqlcmd2 {
 
     }
     process {
+    
+        if(-Not([string]::isNullOrEmpty($OutputFile)) -and $MessagesToOutput) {
+            throw "Parameters OutputFile and MessagesToOutput are mutually exclusive"
```

Comment:
Most probably it can be achieved by using parameter sets

Reply:
I'm not used to parameter sets, but does this means that there should be 4 parameter sets for the function?
Con-Que
Con-Fil
Con-Que-Output
Con-Fil-Output
?

In that case, isn't it too heavy, just for adding one parameter?

Reply:
take a look to the current param sets: [Parameter(ParameterSetName = 'Ins-Que'

Reply:
I think it's too much code addition and duplication (the param zone would double) for just adding one parameter while a 3 line code works well.